modAionImpl/src/org/aion/zero/impl/config/Network.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2017-2018 Aion foundation.
+ *
+ *     This file is part of the aion network project.
+ *
+ *     The aion network project is free software: you can redistribute it
+ *     and/or modify it under the terms of the GNU General Public License
+ *     as published by the Free Software Foundation, either version 3 of
+ *     the License, or any later version.
+ *
+ *     The aion network project is distributed in the hope that it will
+ *     be useful, but WITHOUT ANY WARRANTY; without even the implied
+ *     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *     See the GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+package org.aion.zero.impl.config;
+
+import java.util.Arrays;
+
+/**
+ * Defines different Aion networks.
+ *
+ * @author Alexandra Roatis
+ */
+public enum Network {
+    MAINNET("mainnet", 256),
+    CONQUEST("conquest", 128),
+    MASTERY("mastery", 32),
+    CUSTOM("custom", 0);
+
+    private final String name;
+    private int identifier;
+
+    /**
+     * Constructor.
+     *
+     * @param _name network name
+     * @param _identifier network identifier
+     */
+    Network(String _name, int _identifier) {
+        this.name = _name;
+        this.identifier = _identifier;
+    }
+
+    @Override
+    public String toString() {
+        return this.name;
+    }
+
+    public int getIdentifier() {
+        return identifier;
+    }
+
+    /**
+     * Utility method that determines the correct network based on the given name.
+     *
+     * @param network a string value representing the network name
+     * @return the network object corresponding to the string value or null when the value is not
+     *     mapped to an object.
+     * @implNote There may be several test networks active. The term {@code testnet} is defined to
+     *     correspond to the current default test network.
+     */
+    public static Network determineNetwork(String network) {
+        String netStr = network.toLowerCase();
+
+        if (netStr.equals("testnet")) {
+            return Network.MASTERY;
+        }
+
+        for (Network net : Network.values()) {
+            if (netStr.equals(net.toString())) {
+                return net;
+            }
+        }
+
+        return null;
```

Comment:
would it make more sense to throw IllegalArgumentException?

Reply:
not in this case, because if this method returns `null` the calling method will add the default `mainnet` setting allowing for the kernel to run